projects/amdsmi/src/amd_smi/amd_smi_utils.cc

```diff
@@ -793,6 +793,25 @@ std::string smi_amdgpu_split_string(std::string str, char delim) {
   return "";
 }
 
+// Split string at delimiter and return strings in vector
+std::vector<std::string> split_string(const std::string& line, char delim) {
+  std::vector<std::string> out;
+  std::size_t start = 0;
+
+  while (start < line.size()) {
+    auto pos = line.find(delim, start);
```

Comment:
@adam360x, what happens here if `delim` is not found in the string, and `pos = line.size()`? 
I think that as we have `start = (pos + 1);` it could make `start` to exceed `line.size()` and the loop would continue due to integer overflow/wrap-around?!